lib/shared/addon/components/form-node-selector-term-k8s/component.js

```diff
@@ -78,8 +79,7 @@ export default Component.extend({
     },
     set(key, val){
       if ((!val || !val.length) && this.nodeSelectorTerm.matchExpressions){
-        set(this, 'nodeSelectorTerm.matchExpressions', null)
-
+        delete this.nodeSelectorTerm.matchExpressions
```

Comment:
Technically setting these fields to `null` should work the same as deleting the keys but results in a payload shaped differently than it would in rke2

Reply:
agree